template.yaml

```diff
@@ -0,0 +1,984 @@
+        Format:
+          Fn::ToJsonString:
+            requestId: $context.requestId
+            ip: $context.identity.sourceIp
+            requestTime: $context.requestTime
+            httpMethod: $context.httpMethod
+            path: $context.path"
+            routeKey: $context.routeKey
+            status: $context.status
+            protocol: $context.protocol
+            responseLength: $context.responseLength
+
+  ApiGatewayAccessLogsGroup:
+    Type: AWS::Logs::LogGroup
+    # checkov:skip=CKV_AWS_158:Ensure that CloudWatch Log Group is encrypted by KMS
+    Metadata:
+      checkov:
+        skip:
+          - id: "CKV_AWS_158"
+            comment: "Ensure that CloudWatch Log Group is encrypted by KMS"
+    Properties:
+      LogGroupName: !Sub /aws/apigateway/${AWS::StackName}-ApiGatewayAccessLogsGroup
+      RetentionInDays: 90
+
+  ApiGatewayLambdaFunctionAuthorizer:
+    Type: AWS::Serverless::Function
+    Properties:
+      # checkov:skip=CKV_AWS_116: DLQ not required
+      # checkov:skip=CKV_AWS_117: Internet access not required
+      ReservedConcurrentExecutions: 1
+      Handler: index.handler
+      MemorySize: 128
+      Runtime: nodejs20.x
+      Timeout: 900
+      InlineCode: |
+        'use strict';
+  
+        const ip4ToInt = ip =>
+          ip.split('.').reduce((int, oct) => (int << 8) + parseInt(oct, 10), 0) >>> 0;
+        
+        const isIp4InCidr = ip => cidr => {
+          const [range, bits = 32] = cidr.split('/');
+          const mask = ~(2 ** (32 - bits) - 1);
+          return mask === 0
+            ? ip4ToInt(ip) === ip4ToInt(range)
+            : (ip4ToInt(ip) & mask) === (ip4ToInt(range) & mask);
+        };
+        
+        const isIp4InCidrs = (ip, cidrs) => cidrs.some(isIp4InCidr(ip));
+        
+        exports.handler = async(event) => {
+          const ipAddress = event.requestContext.http.sourceIp;
+          const validIps = [
+            '217.196.229.77/32',
+            '217.196.229.79/32',
+            '217.196.229.80/31',
+            '51.149.8.0/25',
+            '51.149.8.128/29',
+            '213.86.153.211/32',
+            '213.86.153.212/31',
+            '213.86.153.214/32',
+            '213.86.153.235/32',
+            '213.86.153.236/31',
+            '213.86.153.231/32'];
+            const isValidIp = isIp4InCidrs(ipAddress, validIps);
+            return {
+              'isAuthorized': isValidIp
+            };
+        };
+    Tags:
+      - Key: Name
+        Value: !Sub "${AWS::StackName}-ApiGatewayLambdaFunctionAuthorizer"
+      - Key: Service
+        Value: Relying Party stub
+      - Key: Source
+        Value: alphagov/di-auth-stub-relying-party
+      - Key: Owner
+        Value: di-orchestration@digital.cabinet-office.gov.uk
+
+  ApiGatewayAuthorizer:
+    Type: AWS::ApiGatewayV2::Authorizer
+    Properties:
+      Name: !Sub "${AWS::StackName}-ApiGatewayAuthorizer"
+      ApiId: !Ref ApiGateway
+      AuthorizerResultTtlInSeconds: 0
```

Comment:
We can probably add caching to this? Otherwise load tests could get expensive

Reply:
Sure - I wasn't sure what we had to cache based on (something I probably could have thought about a little more)